### DISCUSSION TITLE
Disable Stalebot

### DISCUSSION THREAD

**Discussion by dsbert** [ScoopInstaller/Extras#7094] [General]
Please reconsider the usage of Stalebot.

Either an issue is relevant and should remain open, or it isn't and should be closed. Closing an issue on the basis of recent activity disregards all other reasoning. It also results in spam because every X days a new closure warning is posted and every Y days a new user has to come and bump the issue. 

Here is a recent hackernews thread has a lot more discussion on Stalebot - [GitHub stale bot considered harmful](https://news.ycombinator.com/item?id=28998374)

**Comment by rasa** [comment]
@ScoopInstaller/maintainers Thoughts on this? Perhaps we disable for now, so we have breathing room to discuss before hundreds of issues are closed on us.

**Reply by stevehipwell** [reply]
The context of the Hacker News thread was projects where the issues add value, but I'd argue this isn't always the case. A large number of issues here are throw away and become outdated well before stale bot would pick them up. All triaged issues will get labels to stop stale bot flagging them so the valuable issues are safe. I would argue that anything not triaged should be thrown away as it ends up just being noise. Scoop is a stadium project(s) and needs to be managed as such. #my2cents

**Reply by maphew** [reply]
For 2 months I've no action on a PR I made. Stalebot is telling me it's about to be closed, but the staleness isn't coming from my side: I've not had any feedback on the PR. I'm not bringing this up as a complaint, everyone's time is valueable and their attention moreso. I'm happy to keep bumping it alive until someone has time to look at it,  but I don't wish to be source of unwanted noise either.  If I take this approach I will effectively be a freshbot responding to a stalebot in a recursion loop and maybe no one will come out ahead. _(but it's likely I will tire and quit first ;-)_

**Reply by rasa** [reply]
Perhaps we disable stalebot for PRs only? Per https://github.com/probot/stale#usage , we can do this by adding
```
only: issues
```

**Reply by rashil2000** [reply]
I feel that the bot should remain for now. The manifest repos have had a huge backlog of issues and PRs. And very frankly, the reputation of the entire ScoopInstaller org had been compromised (we all know why) lately. 
Stalebot just helps us get upto speed by bumping PRs so that contributors get reminded that they can (still) have hopes to get their PRs merged.
For issues, a majority of them were hash check failures, and the second-most common premise was that there was some problem in the older version of a manifest, which have been fixed in the newer versions. And as @stevehipwell said, the actual bugs/requests get triaged and labelled appropriately, so Stalebot won't touch them anyway.
Once we clear out the backlog, we can think of restricting Stalebot to issues only, or maybe disable it entirely.

**Reply by niheaven** [reply]
I'd preferred to disable it now and manually pick up valuable issues and PRs, and then reenable it. The vast number of issues and PRs is due to some old reasons and can now be cleaned by active maintainers.

The robot cannot judge value.

**Comment by HUMORCE** [comment]
totally agree with @niheaven.
we still need recheck all of issues/PRs.

**Reply by rasa** [reply]
Both arguments have valid points, but disabling for a bit (a week?), to give us time to manually review and label the keepers, and then enable it again, satisfies both arguments. Sound good?

**Reply by rashil2000** [reply]
If the bot remains active, all the currently `stale` issues/PRs will get closed in 5-6 days. So I say let's disable the bot for now.

**Reply by stevehipwell** [reply]
I agree with disabling the bot short term until we've had a chance to triage everything outstanding, I actually thought the bot started it's timing from its installation. Long term it could be limited to issues only if PRs aren't being automatically generated for hash failures.

**Comment by issaclin32** [comment]
**UPDATE:** The stale bot is temporarily disabled

https://github.com/ScoopInstaller/.github/blob/8be677122fc6f5a80af50ce823138ae30d200a25/.github/stale.yml#L8
```
# Number of days of inactivity before an Issue or Pull Request becomes stale
# Default is 60
daysUntilStale: 99999

# Number of days of inactivity before an Issue or Pull Request with the stale label is closed.
# Set to false to disable. If disabled, issues still need to be closed manually, but will remain marked as stale.
# Default is 7
daysUntilClose: 99999
```
